tools/bpf2c/bpf_code_generator.h

```diff
@@ -92,7 +94,13 @@ class bpf_code_generator
     {
         std::vector<output_instruction_t> output;
         std::set<std::string> referenced_registers;
-        std::string function_name;
+        std::string program_name;
+        GUID program_type = {0};
+        GUID expected_attach_type = {0};
+        // Indices of the maps used in this section.
+        std::set<size_t> referenced_map_indices;
```

Comment:
Why indices and not the names?

Reply:
These are just indices into the global map array to know which maps are used by a specific program. This information is used to associate the used maps to a program.